### ISSUE TITLE
Amplify can't find Gradle6.7 on build phase of CI/CD

### ISSUE THREAD

**Issue by chazarabriseis** [aws-amplify/amplify-console#1243]
I want to deploy an app that is using python and java lambdas.

I got the amplify backend build working by using the docker image wvidana/amplifybuildimage:python38.

To get the amplify backend build for the java environment working I took the above mentioned image and added java and gradle to it and published a new docker image chazarabriseis/amplifybuildimage:python38java11gradle6

Amplify build is now happy with java and it also finds gradle but it complains about the version:
Update Gradle to >=5. Download link: https://bit.ly/3aGYDj6

I don't know why this is the case since I am installing and pointing to verison 6.7 and if asking on a local image build for the gradle verison it is 6.7... Does anyone have an idea why amplify build does not find the gradle version 6.7 or what I am missing?

The changes to the wvidana/amplifybuildimage:python38 Dockerfile are the following

RUN wget https://corretto.aws/downloads/latest/amazon-corretto-11-x64-linux-jdk.tar.gz &&
tar -xf amazon-corretto-11-x64-linux-jdk.tar.gz amazon-corretto-11.0.9.12.1-linux-x64 -C / &&
mv /amazon-corretto-11.0.9.12.1-linux-x64 /usr/bin/java &&
rm -rf amazon-corretto-11-x64-linux-jdk.tar.gz

RUN wget https://services.gradle.org/distributions/gradle-6.7-bin.zip -P /tmp &&
unzip -d /gradle /tmp/gradle-6.7-bin.zip &&
mv /gradle /usr/bin/gradle &&
rm -rf /tmp/gradle-6.7-bin.zip

Environment Setup
RUN echo export PATH="
/usr/local/rvm/gems/ruby-${VERSION_RUBY_DEFAULT}/bin:
/usr/local/rvm/gems/ruby-${VERSION_RUBY_DEFAULT}@global/bin:
/usr/local/rvm/rubies/ruby-${VERSION_RUBY_DEFAULT}/bin:
/usr/bin/java/bin:
/usr/bin/gradle/gradle-6.7/bin:
/usr/local/rvm/bin:
/root/.yarn/bin:
/root/.config/yarn/global/node_modules/.bin:
/root/.nvm/versions/node/${VERSION_NODE_DEFAULT}/bin:
$(python3 -m site --user-base)/bin:
$PATH" >> ~/.bashrc &&
echo export GEM_PATH="/usr/local/rvm/gems/ruby-${VERSION_RUBY_DEFAULT}" >> ~/.bashrc &&
echo "nvm use ${VERSION_NODE_DEFAULT} 1> /dev/null" >> ~/.bashrc

**Comment by kahdojay** [comment]
@chazarabriseis can you provide your appId, region, branch name, and example build number so we can take a closer look?

**Comment by chazarabriseis** [comment]
@kahdojay thanks for looking in to it! 
Here the info for the failed build:
"AmplifyAppId": "d15zxau0438qdt"
"Region": "eu-central-1",
Branch name: master,
build number: 73

I was able to get a successful build just then, build number 72, by adding the installation to the amplify.yaml file. This hack would mean it downloads it every build time... so I guess it would be better and cleaner if it would work with the docker image?

This is what I added to the phases: build: commands: in the amplify.yaml file to make it recognize the correct gradle version:
        - wget https://services.gradle.org/distributions/gradle-6.3-bin.zip -P /tmp
        - unzip -d /gradle /tmp/gradle-*.zip
        - mv /gradle /usr/bin/gradle
        - echo export PATH="/usr/bin/gradle/gradle-6.7/bin:$PATH" >> ~/.bashrc
        - gradle --version
